generator/src/test/java/com/scottlogic/deg/generator/cucumber/utils/InMemoryOutputTarget.java

```diff
@@ -0,0 +1,56 @@
+package com.scottlogic.deg.generator.cucumber.utils;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.ProfileFields;
+import com.scottlogic.deg.generator.outputs.GeneratedObject;
+import com.scottlogic.deg.generator.outputs.targets.OutputTarget;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Defines an output target which stores the output data into the test state.
+ */
+public class InMemoryOutputTarget implements OutputTarget {
+
+    private TestState testState;
+
+    @Inject
+    public InMemoryOutputTarget(TestState testState) {
+        this.testState = testState;
+    }
+
+    @Override
+    public void outputDataset(Stream<GeneratedObject> generatedObjects, ProfileFields profileFields) throws IllegalStateException{
+        this.testState.generatedObjects = getRows(generatedObjects);
+    }
+
+    //TODO: Don't Want This
```

Comment:
Todo should likely be removed

Reply:
Good spot, will investigate if this can be done yet or just accepted.